types/header.d.ts

```diff
@@ -2,3 +2,155 @@
  * The header type declaration of `undici`.
  */
 export type IncomingHttpHeaders = Record<string, string | string[] | undefined>;
+
+type HeaderNames =
+  | 'Accept'
+  | 'Accept-CH'
+  | 'Accept-Charset'
+  | 'Accept-Encoding'
+  | 'Accept-Language'
+  | 'Accept-Patch'
+  | 'Accept-Post'
+  | 'Accept-Ranges'
+  | 'Access-Control-Allow-Credentials'
+  | 'Access-Control-Allow-Headers'
+  | 'Access-Control-Allow-Methods'
+  | 'Access-Control-Allow-Origin'
+  | 'Access-Control-Expose-Headers'
+  | 'Access-Control-Max-Age'
+  | 'Access-Control-Request-Headers'
+  | 'Access-Control-Request-Method'
+  | 'Age'
+  | 'Allow'
+  | 'Alt-Svc'
+  | 'Alt-Used'
+  | 'Authorization'
+  | 'Cache-Control'
+  | 'Clear-Site-Data'
+  | 'Connection'
+  | 'Content-Disposition'
+  | 'Content-Encoding'
+  | 'Content-Language'
+  | 'Content-Length'
+  | 'Content-Location'
+  | 'Content-Range'
+  | 'Content-Security-Policy'
+  | 'Content-Security-Policy-Report-Only'
+  | 'Content-Type'
+  | 'Cookie'
+  | 'Cross-Origin-Embedder-Policy'
+  | 'Cross-Origin-Opener-Policy'
+  | 'Cross-Origin-Resource-Policy'
+  | 'Date'
+  | 'Device-Memory'
+  | 'ETag'
+  | 'Expect'
+  | 'Expect-CT'
+  | 'Expires'
+  | 'Forwarded'
+  | 'From'
+  | 'Host'
+  | 'If-Match'
+  | 'If-Modified-Since'
+  | 'If-None-Match'
+  | 'If-Range'
+  | 'If-Unmodified-Since'
+  | 'Keep-Alive'
+  | 'Last-Modified'
+  | 'Link'
+  | 'Location'
+  | 'Max-Forwards'
+  | 'Origin'
+  | 'Permissions-Policy'
+  | 'Priority'
+  | 'Proxy-Authenticate'
+  | 'Proxy-Authorization'
+  | 'Range'
+  | 'Referer'
+  | 'Referrer-Policy'
+  | 'Retry-After'
+  | 'Sec-Fetch-Dest'
+  | 'Sec-Fetch-Mode'
+  | 'Sec-Fetch-Site'
+  | 'Sec-Fetch-User'
+  | 'Sec-Purpose'
+  | 'Sec-WebSocket-Accept'
+  | 'Server'
+  | 'Server-Timing'
+  | 'Service-Worker-Navigation-Preload'
+  | 'Set-Cookie'
+  | 'SourceMap'
+  | 'Strict-Transport-Security'
+  | 'TE'
+  | 'Timing-Allow-Origin'
+  | 'Trailer'
+  | 'Transfer-Encoding'
+  | 'Upgrade'
+  | 'Upgrade-Insecure-Requests'
+  | 'User-Agent'
+  | 'Vary'
+  | 'Via'
+  | 'WWW-Authenticate'
+  | 'X-Content-Type-Options'
+  | 'X-Frame-Options'
+  | (string & {})
```

Comment:
Maybe create a utility.d.ts if not exisiting already and create a Autocomplete type?